src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_ReadOnlyStruct.cs

```diff
@@ -0,0 +1,675 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.CSharp.UnitTests;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using System.Collections.Immutable;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    [CompilerTrait(CompilerFeature.ReadOnlyReferences)]
+    public class ReadOnlyStruct : CSharpTestBase
+    {
+        [Fact]
+        public void IsReadOnlyIsWrittenToMetadata_SameAssembly()
+        {
+            var text = @"
+namespace System.Runtime.CompilerServices
+{
+    public class IsReadOnlyAttribute : System.Attribute { }
+}
+class Test
+{
+    public readonly struct S1 {}
+}
+";
+
+            CompileAndVerify(text, verify: false, symbolValidator: module =>
+            {
+                var type = module.ContainingAssembly.GetTypeByMetadataName("Test").GetTypeMember("S1");
+                Assert.True(type.IsReadOnly);
+
+                AssertReferencedIsReadOnlyAttribute(Accessibility.Public, type.GetAttributes(), module.ContainingAssembly.Name);
```

Comment:
We have much more extensive tests for IsReadOnlyAttribute and the whole embedding mechanism as a part of "ref readonly" since that feature started using this attribute first. 

In this change I am only hooking up into that implementation form one more place - when struct is `readonly` . I think it is sufficient to test that hookup is successful - i.e. it causes the attribute usage or triggers generation when necessary. The new tests provide enough coverage for that.

---
In reply to: [119923620](https://github.com/dotnet/roslyn/pull/19888#discussion_r119923620) [](ancestors = 119923620)